src/swarm-lang/Swarm/Language/Parser/QQ.hs

```diff
@@ -40,3 +43,27 @@ quoteTypeExp s = do
         )
   parsed <- runParserTH pos (fully sc parsePolytype) s
   dataToExpQ (fmap liftText . cast) parsed
+
+astQ :: QuasiQuoter
+astQ =
+  QuasiQuoter
+    { quoteExp = quoteASTExp
+    , quotePat = error "quotePat  not implemented for ASTs"
+    , quoteType = error "quoteType not implemented for ASTs"
+    , quoteDec = error "quoteDec  not implemented for ASTs"
+    }
+
+quoteASTExp :: String -> TH.ExpQ
+quoteASTExp s = do
+  loc <- TH.location
+  let pos =
+        ( TH.loc_filename loc
+        , fst (TH.loc_start loc)
+        , snd (TH.loc_start loc)
+        )
+  parsed <- runParserTH pos (fully sc parseTerm) s
```

Comment:
Why does `runParserTH` not take TH location directly?

Reply:
That... is a good question!